lib/vmdb/util.rb

```diff
@@ -1,7 +1,17 @@
 module VMDB
   module Util
-    def self.http_proxy_uri
-      proxy = VMDB::Config.new("vmdb").config[:http_proxy] || {}
+    def self.http_proxy_uri(proxy_config = :default)
+      # TODO: (julian) This looks really messy. Need to make it look nicer.
+      vmdb_proxy = VMDB::Config.new("vmdb").config[:http_proxy]
+
+      return nil unless vmdb_proxy
+
+      if vmdb_proxy[proxy_config].nil?
+        $log.warn("Could not find proxy setting for #{proxy_config}")
+      end
+
+      proxy = vmdb_proxy[proxy_config] || {}
```

Comment:
If this line is changed to
```ruby
proxy = vmdb_proxy[proxy_config] || vmdb_proxy
```

you make sure that former configurations still work